syntax="proto3";

package economy;

message ProtoDate {
    int32 day = 1;
    int32 month = 2;
    int32 year = 3;
}

message DataPayload {
    ProtoDate date = 1;
    float value = 2;
}

message DataRequest {
    ProtoDate begin = 1;
    ProtoDate end = 2;
}

message DataReply {
    repeated DataPayload content = 1;
}

enum Currency {
    RON = 0;
    EUR = 1;
    USD = 2;
}

message CurrencyRequest {
    Currency new_currency = 1;
}

message CurrencyReply {
    double conversion_rate = 1;
}


service TransportService {
    rpc GetData(DataRequest) returns (DataReply) {}
    rpc ChangeCurrency(CurrencyRequest) returns (CurrencyReply) {}
}
